### ISSUE TITLE
Task handling

### ISSUE THREAD

**Issue by vchuravy** [jrevels/Cassette.jl#120]
Right now using a Task allows one to escape Cassette :(
In the below example the overdub of `g` is not triggered.

```julia
using Cassette
g() = "Nah"
Cassette.@context Ctx
Cassette.overdub(::Ctx, ::typeof(g)) = "Yah"
f() = @sync begin
             @async @show g()
         end
Cassette.overdub(Ctx(), f)
```

**Comment by vchuravy** [comment]
Due to a quirk in the current implementation of `Task`, the actual constructor is `_Task`.
So one can define:
```
Cassette.overdub(ctx::Ctx, ::typeof(Base.Core._Task), @nospecialize(f), stack::Int, future) = Base.Core._Task(()->Cassette.overdub(ctx, f), stack, future)
```

@jrevels this raised a broader point. All anonymous functions have the possibility to escape out of a Context if they are being passed outsides Cassette control (here it is `:jl_new_task`).

**Comment by jrevels** [comment]
ref #77 

Seems like Cassette would need to hook into the scheduler or something for us to tackle concurrency issues, yeah? At that point, I wonder whether it'd just be easier/more sane to actually have Cassette implemented in the compiler itself. Hmm...

**Comment by vchuravy** [comment]
Agreed the semantics of metadata in the context of parallelism is non-trivial and needs careful design. What I need is contextual execution to work with task, so that once can have a reasonable closed world assumption, e.g. no way to escape the context.
